### PR TITLE
Fix compilation for Xcode 16

### PR DESCRIPTION
**Pull request by guykogus**
Also includes SwiftLint fixes

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.6.
- [x] New extensions support iOS 12.0+ / tvOS 12.0+ / macOS 10.13+ / watchOS 4.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

**Comment by guykogus** [comment]
@LucianoPAlmeida maybe I'll separate out the SwiftLint fixes. Will get to it when I can.

**Comment by guykogus** [comment]
@LucianoPAlmeida WDYT? I'll leave Swift 6 fixes for the next PR

File: SwifterSwift.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -3572,7 +3566,6 @@
 				GCC_WARN_UNUSED_VARIABLE = YES;
 				INFOPLIST_FILE = "$(SRCROOT)/Sources/Info.plist";
 				INSTALL_PATH = "$(LOCAL_LIBRARY_DIR)/Frameworks";
-				IPHONEOS_DEPLOYMENT_TARGET = 10.0;
```
Comment (LucianoPAlmeida) [inline SwifterSwift.xcodeproj/project.pbxproj:3575]: I think it may be time to up the minimum to 12 or 13 maybe? But that is something for another PR.

**Review by LucianoPAlmeida — APPROVED** [review]

File: SwifterSwift.xcodeproj/project.pbxproj (hunk 1 above)
Comment (guykogus) [reply SwifterSwift.xcodeproj/project.pbxproj:3575]: Is there any reason to do it? I don't think the new Xcode complains about the minimum being 12.

Comment (LucianoPAlmeida) [reply SwifterSwift.xcodeproj/project.pbxproj:3575]: It doesn't complain but I think right now our minimum is 10, so it feels a good time up it to 12 or 13

Comment (guykogus) [reply SwifterSwift.xcodeproj/project.pbxproj:3575]: We're on iOS 12, which I think is fine for now.

**Pull request merged**